ios/FluentUI/Controls/IndeterminateProgressBarView.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+public typealias MSIndeterminateProgressBarView = IndeterminateProgressBarView
+
+/// Indeterminate progress bar. Since iOS's UIprogressView is always determinate, this is created for handling indeterminate scenarios.
+@objc(MSFIndeterminateProgressBarView)
+open class IndeterminateProgressBarView: UIView {
+
+    private struct Constants {
+        static let animationDuration: TimeInterval = 2.5
+        static let animationKey: String = "progressAnimation"
+        static let animationTimingFunction = CAMediaTimingFunction(controlPoints: 0.28, 0.08, 0.72, 0.92)
+        static let opacityKeyFrame: [NSNumber] = [0, 0.375, 0.625, 1]
+        static let opacityValues: [NSNumber] = [0.0, 1.0, 1.0, 0.0]
+        static let progressBarHeight: CGFloat = 2
+    }
+
+    /// progressbar's track
+    private var trackLayer = CALayer()
+
+    /// gradient layer that slides across the track
+    private var gradientLayer = CAGradientLayer()
+
+    /// The grouped animations that will be applied to the gradient layer
+    private var animationGroup = CAAnimationGroup()
+
+    /// The progress bar view should be hidden when animation stops if set to true
+    @objc open var hidesWhenStopped: Bool = true
+
+    // Don't modify this directly. Instead, call `startAnimating` and `stopAnimating`
+    @objc(isAnimating) public private(set) var isAnimating: Bool = false
+
+    open override var intrinsicContentSize: CGSize {
+        return CGSize(width: UIView.noIntrinsicMetric, height: frame.height)
+    }
+
```

Comment:
you would need to override size that fits as well..
    open override func sizeThatFits(_ size: CGSize) -> CGSize {
        return CGSize(width: size.width, height: Constants.progressBarHeight)
    }